Makefile.in

```diff
@@ -172,11 +173,29 @@ uninstall:
 	@rm -rf -v --preserve-root=all $(DESTDIR)$(DATADIR)/$(INSTALLDIR)
 
 
-.PHONY: user-install
-user-install:
-	$(info :: --- )
-	$(info :: Installed under user-only location "$(DATADIR)/$(INSTALLDIR)" )
-	$(info :: To run you need to make sure "$(BINDIR)" is in your PATH )
+ZIPAPP := $(OBJDIR)/umu-run
+ZIPAPP_STAGING := $(OBJDIR)/zipapp_staging
+ZIPAPP_VENV := $(OBJDIR)/zipapp_venv
+
+$(OBJDIR)/.build-zipapp: | $(OBJDIR) version
+	$(info :: Building umu-launcher as zipapp )
+	$(PYTHON_INTERPRETER) -m venv $(ZIPAPP_VENV)
+	. $(ZIPAPP_VENV)/bin/activate && python3 -m pip install -t "$(ZIPAPP_STAGING)" -U --no-compile .
+	install -Dm644 umu/umu_version.json "$(ZIPAPP_STAGING)"/umu/umu_version.json
+	cp umu/__main__.py "$(ZIPAPP_STAGING)"
+	# Make zipapp reproducible
+	find "$(ZIPAPP_STAGING)" -exec touch -h -d "$(SOURCE_DATE_EPOCH)" {} +
+	. $(ZIPAPP_VENV)/bin/activate && python3 -m zipapp $(ZIPAPP_STAGING) -o $(ZIPAPP) -p "$(PYTHON_INTERPRETER)" -c
+	touch -d "$(SOURCE_DATE_EPOCH)" $(ZIPAPP)
+	touch $(@)
+
+.PHONY: zipapp
+zipapp: $(OBJDIR)/.build-zipapp
 
+zipapp-install: zipapp
+	$(info :: Installing umu-launcher as zipapp )
+	install -d $(DESTDIR)$(PREFIX)/bin
+	install -Dm755 $(ZIPAPP) $(DESTDIR)$(PREFIX)/bin/$(ZIPAPP)
```

Comment:
I have made a mistake here, `$(PREFIX)/bin` could also be referenced as `$(BINDIR)`.